lib/libutee/tee_api.c

```diff
@@ -754,12 +754,16 @@ static bool addr_is_in_no_share_heap(void *p)
 void *TEE_Realloc(void *buffer, size_t newSize)
 {
 	if (!newSize) {
-		TEE_Free(buffer);
+		void *ret = NULL;
 
 		if (addr_is_in_no_share_heap(buffer))
```

Comment:
For the record, this is a false positive. But this may look a bit odd so I believe we're better off checking the pointer first.